### ISSUE TITLE
Ubuntu 17.10 box configured with :private_network, {:type=>"dhcp"} gets wrong address with netplan

### ISSUE THREAD

**Issue by tolland** [hashicorp/vagrant#9570]
### Vagrant version
```
$ vagrant -v
Vagrant 2.0.2
```

### Host operating system
```
$ cat /etc/fedora-release
Fedora release 27 (Twenty Seven)
```

### Guest operating system
```
# lsb_release -d
Description:    Ubuntu 17.10
```

### Vagrantfile
```ruby
Vagrant.configure("2") do |c|
  c.berkshelf.enabled = false if Vagrant.has_plugin?("vagrant-berkshelf")
  if Vagrant.has_plugin?("vagrant-cachier")
    c.cache.scope = :machine
   end
  c.vm.box = "bento/ubuntu-17.10"
  c.vm.hostname = "default-ubuntu-1710.node-red.limepepper.testbox"
  c.vm.network(:private_network, {:type=>"dhcp"})
  c.vm.synced_folder ".", "/vagrant", disabled: true
  c.vm.synced_folder "/home/user1/.kitchen/cache", "/tmp/omnibus/cache", create: true
  c.vm.provider :virtualbox do |p|
  end
end
```

### Debug output
https://gist.github.com/anonymous/1b5fb2d51bb4b1e0da48872f00fd5a10

### Expected behavior

I am adding a private dhcp network using the following config.

    c.vm.network(:private_network, {:type=>"dhcp"})

I expect the machine/interface to be assigned an ip that was offered by the DHCP server. The logs are showing DHCP requests and replies as expected;
```
Mar 14 00:20:55 vagrant dhclient[18843]: DHCPREQUEST of 172.28.128.5 on eth1 to 172.28.128.2 port 67 (xid=0x35b41cbd)
Mar 14 00:20:55 vagrant dhclient[18843]: DHCPACK of 172.28.128.5 from 172.28.128.2
Mar 14 00:20:56 vagrant dhclient[18843]: bound to 172.28.128.5 -- renewal in 481 seconds.
```

(I a using this config on several machines, including debian-9, CentOS-7, CentOS-6, and only see this behaviour on Ubuntu 1710 hosts)

This should assign a DHCP address from VirtualBox, using this dhcp server;
`/usr/lib64/virtualbox/VBoxNetDHCP --ip-address 172.28.128.2 --lower-ip 172.28.128.3 --mac-address 08:00:27:63:67:38 --netmask 255.255.255.0 --network HostInterfaceNetworking-vboxnet4 --trunk-name vboxnet4 --trunk-type netflt --upper-ip 172.28.128.254`

### Actual behavior

Ip address is incorrect...

```
root@default-ubuntu-1710:~# ip addr show dev eth1
4: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:6c:33:22 brd ff:ff:ff:ff:ff:ff
    inet 172.28.128.1/24 brd 172.28.128.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe6c:3322/64 scope link
       valid_lft forever preferred_lft forever
```
Ip address is being set by netplan, in a file generated by vagrant

```
root@default-ubuntu-1710:~# cat /etc/netplan/99-vagrant.yaml
---
network:
  version: 2
  renderer: networkd
  ethernets:
    eth1:
      addresses:
      - 172.28.128.1/24
```

**Comment by WIgor** [comment]
Ubuntu 18.04 - has same behaviour

**Comment by WIgor** [comment]
Looks like this behavior is a combination of these two lines:

https://github.com/hashicorp/vagrant/blob/master/plugins/providers/virtualbox/action/network.rb#L260

and 

https://github.com/hashicorp/vagrant/blob/master/plugins/guests/debian/cap/configure_networks.rb#L40

**Comment by chrisroberts** [comment]
Fixed in #9855 which has been merged. Cheers!
